python-frontend/src/main/java/org/sonar/python/semantic/v2/types/Assignment.java

```diff
@@ -19,9 +19,86 @@
  */
 package org.sonar.python.semantic.v2.types;
 
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 import org.sonar.plugins.python.api.tree.Expression;
 import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.Tree;
 import org.sonar.python.semantic.v2.SymbolV2;
+import org.sonar.python.semantic.v2.UsageV2;
+import org.sonar.python.tree.NameImpl;
+import org.sonar.python.types.HasTypeDependencies;
+import org.sonar.python.types.v2.PythonType;
+import org.sonar.python.types.v2.UnionType;
 
-public record Assignment(SymbolV2 lhsSymbol, Name lhsName, Expression rhs) {
+public class Assignment {
+
+  final SymbolV2 lhsSymbol;
+  Name lhsName;
+  Expression rhs;
+  Set<SymbolV2> variableDependencies = new HashSet<>();
```

Comment:
I'd prefer to have a single place to initialize everything. If you have a constructor - may be it makes sense to initialize these fields there as well

Reply:
If you don't mind I'll address this as part of SONARPY-1826.